### PR TITLE
feat: add version decoration

### PR DESCRIPTION
**Pull request by JoeZwet**
image
adds `+build.<branch>.<commit>` to the version.
The branch is not included if it is `main`
This allows for better debugging when players come in with their self-compiled versions asking why something isn't working.\
Set the `RELEASE` environment var to build without the decoration.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -250,3 +250,32 @@ license {
...
     }
 }
+
+// inspired by https://github.com/TerraformersMC/GradleScripts/blob/2.0/ferry.gradle
+fun getVersionDecoration(): String {
+    if(project.hasProperty("release")) return ""
+
+    var version = "+build"
+    val branch = "git branch --show-current".execute()
+    if(branch.isNotEmpty() && branch != "main") {
```
Comment (marcus8448) [inline build.gradle.kts:260]:
```suggestion
    if(branch.isNotEmpty() && branch.equals("main")) {
```

**Review by marcus8448 — CHANGES_REQUESTED** [review]
I'm assuming that == and #equals works the same as java.

File: build.gradle.kts (hunk 1 above)
Comment (JoeZwet) [reply build.gradle.kts:260]: wouldnt it be `!branch.equals("main")`?

**Comment by JoeZwet** [comment]
current build error:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/workspace/build.gradle.kts' line: 277

* What went wrong:
Cannot run program "git" (in directory "/workspace"): error=2, No such file or directory
> error=2, No such file or directory
```
trying to find a fix

File: build.gradle.kts (hunk 1 above)
Comment (JoeZwet) [reply build.gradle.kts:260]: tho intellij seems to think that `!=` is better 
image

Comment (marcus8448) [reply build.gradle.kts:260]: Ok, I assume kotlins equals handling is different than java then.

**Review by marcus8448 — COMMENTED** [review]
Should there be an override for whatever build system we'll be using? If we don't have a build number before the `+` no version parser will be able to tell if `d72dc71` is newer than `f4eb290`.

File: build.gradle.kts [hunk 2]
```diff
@@ -250,3 +250,33 @@ license {
...
+// from https://discuss.gradle.org/t/how-to-run-execute-string-as-a-shell-command-in-kotlin-dsl/32235/5
+fun String.execute(workingDir: File = project.file("./")): String {
+    val parts = this.split("\\s".toRegex())
+    val proc = ProcessBuilder(*parts.toTypedArray())
+        .directory(workingDir)
+        .redirectOutput(ProcessBuilder.Redirect.PIPE)
+        .redirectError(ProcessBuilder.Redirect.PIPE)
+        .start()
+
+    proc.waitFor(1, TimeUnit.MINUTES)
```
[the remark this answers is not in the document]
Comment (JoeZwet) [reply build.gradle.kts:280]: iirc just outputs as is

**Review by marcus8448 — APPROVED** [review]

**Pull request merged**